tests/integration/test_with_portainer_service.py

```diff
@@ -21,9 +88,265 @@ async def test_portainer_connection(
     event_loop: asyncio.AbstractEventLoop,
     portainer_container: tuple[URL, str],
     aiohttp_client_session: ClientSession,
-):
+) -> None:
     portainer_url, portainer_password = portainer_container
 
     await portainer.authenticate(
         portainer_url, aiohttp_client_session, "admin", portainer_password
     )
+
+
+@pytest.fixture
+async def portainer_endpoint_id(
+    event_loop: asyncio.AbstractEventLoop,
+    portainer_container: tuple[URL, str],
+    aiohttp_client_session: ClientSession,
+    portainer_bearer_code: str,
+) -> int:
+    portainer_url, _ = portainer_container
+
+    endpoint = await portainer.get_first_endpoint_id(
+        portainer_url, aiohttp_client_session, portainer_bearer_code
+    )
+    assert type(endpoint) == int
+    return endpoint
+
+
+async def test_portainer_test_create_stack(
+    portainer_container: tuple[URL, str],
+    aiohttp_client_session: ClientSession,
+    portainer_bearer_code: str,
+    portainer_endpoint_id: int,
+    valid_docker_stack: ComposeSpecsDict,
+    docker_swarm: None,
+    stack_name,
+    clean_stack,
+) -> None:
+    portainer_url, _ = portainer_container
+    ## Assert that formating to URL does not throw:
+    try_to_format_url: URL = URL(portainer_url)
+    assert try_to_format_url
+    #
+    swarm_id = await portainer.get_swarm_id(
+        portainer_url,
+        aiohttp_client_session,
+        portainer_bearer_code,
+        portainer_endpoint_id,
+    )
+
+    await portainer.post_new_stack(
+        base_url=portainer_url,
+        app_session=aiohttp_client_session,
+        bearer_code=portainer_bearer_code,
+        swarm_id=swarm_id,
+        endpoint_id=portainer_endpoint_id,
+        stack_name=stack_name,
+        stack_cfg=valid_docker_stack,
+    )
+    # Wait for the stack to be present
+    async for attempt in AsyncRetrying(
+        stop=stop_after_attempt(10),
+        wait=wait_fixed(1),
+    ):
+        with attempt:
+            await portainer.get_current_stack_id(
+                base_url=portainer_url,
+                app_session=aiohttp_client_session,
+                bearer_code=portainer_bearer_code,
+                stack_name=stack_name,
+            )
```

Comment:
minor: this could have been a function since you use it in multiple places, that way it's simpler to tune the stop and wait parameters if it is require in the future.